guides/source/active_record_aggregations.md

```diff
@@ -0,0 +1,350 @@
+product.price_currency                  # => "USD"
+product.price                           # => Money value object
+
+# Updating price
+product.price = Money.new(2000, "EUR")
+product.price.amount                    # => 2000
+product.price.currency                  # => "EUR"
+product.price_cents                     # => 2000
+product.price_currency                  # => "EUR"
+
+# assigning non-Money value, it will use the converter proc to convert it to Money object
+product.price = 3000
+product.price.amount                    # => 3000
+product.price.currency                  # => "USD" (default currency from constructor)
+product.price_cents                     # => 3000
+product.price_currency                  # => "USD"
+product.price.exchange_to("EUR")        # => Money.new(2550, "EUR")
+
+# comparing prices
+product.price > Money.new(1500, "USD")  # => true
+product.price > Money.new(2700, "EUR")  # => false (1 EUR == 1.17 USD)
+product.price == Money.new(3000, "USD") # => true
+```
+
+### Using with Other Value Objects
+
+You can use `composed_of` with any value object, not just `Money`.
+For example, let's say you have a `User` model that has `street`, `city`, `state`, and `zip` columns in the database,
+and you want to use an `Address` value object to represent the user's address.
+You can define the `User` model like this:
+
+```ruby
+
+class User < ApplicationRecord
+  composed_of :address,
+              class_name: "Address",
+              mapping: [%w(street street), %w(city city), %w(state state), %w(zip zip)],
+              constructor: Proc.new { |street, city, state, zip| Address.new(street || "", city || "", state || "", zip || "") },
+              converter: Proc.new { |value| value.is_a?(Address) ? value : Address.new(value) }
+end
+```
+
+Now you can use the `address` attribute of the `User` model as an `Address` object:
+
+```ruby
+user = User.new(address: Address.new("123 Main St", "Springfield", "IL", "62701"))
+user.address.street # => "123 Main St"
+user.address.city   # => "Springfield"
+user.address.state  # => "IL"
+user.address.zip    # => "62701"
+user.street         # => "123 Main St"
+user.city           # => "Springfield"
+user.state          # => "IL"
+user.zip            # => "62701"
+
+user.address = Address.new("456 Elm St", "Springfield", "IL", "62702")
+user.address.street # => "456 Elm St"
+user.address.city   # => "Springfield"
+user.address.state  # => "IL"
+user.address.zip    # => "62702"
+user.street         # => "456 Elm St"
+user.city           # => "Springfield"
+user.state          # => "IL"
+user.zip            # => "62702"
+```
+
+Querying Aggregated Attributes
+-----------------------------------------------
+
+You can make simple queries using the aggregated attributes.
+When you query using an aggregated attribute, Active Record will translate the query to use the underlying
+table columns.
+
+```ruby
+expensive_products  = Product.where("price_cents > ?", 1000)
```

Comment:
Shouldn't we be using the Money object here?

Reply:
yes we should. done!